### PR TITLE
clean highlightSourceLines code

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
This is the first part of https://github.com/rust-lang/rust/issues/66046. Now that I've splitted the hashchange stuff and the source code lines highlighting, I'll be able to fix the whole issue once and for all.

r? @kinnison

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -163,59 +163,71 @@ function getSearchElement() {
...
-            });
-            for (i = from; i <= to; ++i) {
-                elem = document.getElementById(i);
-                if (!elem) {
-                    break;
-                }
-                addClass(elem, "line-highlighted");
-            }
-        } else if (ev !== null && search && !hasClass(search, "hidden") && ev.newURL) {
+            return highlightSourceLines(match, ev);
```
Comment (kinnison) [inline src/librustdoc/html/static/main.js]: If you simply called this unconditionally, passing `undefined` for `match` then you wouldn't need to duplicate the match expression in this function and in `hilightSourceLines` -- also you'd be able to eliminate the duplication of the call to `hideSidebar()`

Comment (kinnison) [reply src/librustdoc/html/static/main.js]: You could have `highlightSourceLines()`  return whether it did work, and use that to decide if you want to show the search element.

**Review by kinnison — COMMENTED** [review]
One possible refactor suggestion.  I'll have to take your word for it that this works though since I'm not near a test system right now.

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustdoc/html/static/main.js]: I'm not a big fan of the idea... :-/ If it matches, then we shouldn't look anywhere else. The code is more clear from my point of view in this way. Also, what would the boolean mean? If we encounter an error while trying to highlight the lines, should we return `true` or `false`? :)

Comment (kinnison) [reply src/librustdoc/html/static/main.js]: Okay that's fair, you can resolve this comment without change.

**Review by kinnison — APPROVED** [review]
With @GuillaumeGomez 's raised concern, I withdraw my refactor suggestion.

I have not been able to confirm this still works though, so that's on the author :D

**Comment by GuillaumeGomez** [comment]
I can confirm it works locally so let's go!

@bors: r=kinnison rollup

**Pull request merged**